publisher/publisher.go

```diff
@@ -6,273 +6,138 @@
 package publisher
 
 import (
-	"bytes"
-	"crypto/ecdsa"
-	"crypto/sha256"
 	"crypto/x509"
 	"encoding/base64"
-	"encoding/json"
 	"fmt"
-	"io/ioutil"
 	"net/http"
-	"strconv"
 	"strings"
-	"time"
+
+	ct "github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/google/certificate-transparency/go"
+	ctClient "github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/google/certificate-transparency/go/client"
 
 	"github.com/letsencrypt/boulder/core"
 	blog "github.com/letsencrypt/boulder/log"
-	"github.com/letsencrypt/boulder/sa"
 )
 
-// LogDescription tells you how to connect to a log and verify its statements.
-type LogDescription struct {
-	ID        string
-	URI       string
-	PublicKey *ecdsa.PublicKey
-}
-
-type rawLogDescription struct {
-	URI       string `json:"uri"`
-	PublicKey string `json:"key"`
+// Log contains the CT client and signature verifier for a particular CT log
+type Log struct {
+	Client   *ctClient.LogClient
```

Comment:
These fields can be unexported because they are only used from within the package.